### PR TITLE
[7770] Upgrade to ruby 3.3

### PR DESCRIPTION
**Pull request by defong**
### Context
Ruby 3.3 has been out n the wild for sometime now... it be good to tame it.

### Changes proposed in this pull request
Use ruby 3.3

### Guidance to review

Nothing breaks

### Important business

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [ ] Do we need to send any updates to DQT as part of the work in this PR?
- [ ] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

**Comment by defong** [comment]
[Upgrade to Ruby 3.3](https://trello.com/c/r0APi4LQ/7770-upgrade-to-ruby-33)

File: .github/actions/smoke-test/action.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ runs:
     - name: Set up Ruby 3.2.5
```
Comment (tomtrentham) [inline .github/actions/smoke-test/action.yml:28]: Should probably keep this in sync.

Comment (defong) [reply .github/actions/smoke-test/action.yml]: Cool I change it be agnostic

File: .github/actions/smoke-test/action.yml [hunk 2]
```diff
@@ -25,10 +25,10 @@ inputs:
 runs:
   using: composite
   steps:
-    - name: Set up Ruby 3.2.5
+    - name: Setup Ruby
```
Comment (tomtrentham) [inline .github/actions/smoke-test/action.yml:28]: I think we want the verb here rather than the noun
```suggestion
    - name: Set up Ruby
```

**Review by tomtrentham — APPROVED** [review]
As long as Steve's happy

**Pull request merged**